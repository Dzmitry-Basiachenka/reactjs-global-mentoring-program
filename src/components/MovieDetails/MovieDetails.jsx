import React, { useEffect } from 'react';
import { Container, Row, Col } from 'react-bootstrap';

import { getGenres, getYear, getRuntime } from '../../utils/utils.js';
import styles from './styles.module.css';

const MovieDetails = ({ movie }) => {
    return (
        <Container>
            <Row>
                <Col md={3}>
                    <img data-testid='movie-poster' className="img-fluid" src={movie.poster_path} alt={movie.title} />
                </Col>
                <Col md={9}>
                    <div className={styles.flexbox}>
                        <span data-testid='movie-title' className={styles.title}>{movie.title}</span>
                        {movie.vote_average && <span data-testid='movie-vote' className={styles.vote}>{movie.vote_average}</span>}
                    </div>
                    <p data-testid='movie-tagline'>{movie.tagline}</p>
                    {movie.genres && <p data-testid='movie-genres'>{getGenres(movie.genres)}</p>}
                    <p className={styles.flexbox}>
                        {movie.release_date && <span data-testid='movie-year' className={styles.year}>{getYear(movie.release_date)}</span>}
                        {movie.runtime && <span data-testid='movie-runtime' className={styles.runtime}>{getRuntime(movie.runtime)}</span>}
                    </p>
                    <p data-testid='movie-overview'>{movie.overview}</p>
                </Col>
            </Row>
        </Container>
    )
}

export default MovieDetails;
