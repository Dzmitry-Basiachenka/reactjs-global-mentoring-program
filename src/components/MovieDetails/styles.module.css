.flexbox {
    display: flex;
    align-items: center;
}

.title {
    font-size: 32px;
    font-weight: 500;
}

.vote {
    margin-left: 20px;
    border: 1px solid #1e7e34;
    border-radius: 20px;
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: 20px;
    flex-shrink: 0;
}

.year,
.runtime {
    color: #f65261;
    font-size: 24px;
    margin-right: 50px;
}
