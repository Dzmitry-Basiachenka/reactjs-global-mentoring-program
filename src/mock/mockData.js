export const MOCK_MOVIES = [
  {
    "id": 680,
    "title": "Pulp Fiction",
    "tagline": "Just because you are a character doesn't mean you have character.",
    "vote_average": 8.3,
    "vote_count": 10910,
    "release_date": "1994-09-10",
    "poster_path": "https://cdn.europosters.eu/image/1300/posters/pulp-fiction-cover-i1288.jpg",
    "overview": "A burger-loving hit man, his philosophical partner, a drug-addled gangster's moll and a washed-up boxer converge in this sprawling, comedic crime caper. Their adventures unfurl in three stories that ingeniously trip back and forth in time.",
    "budget": 8000000,
    "revenue": 213928762,
    "genres": [
      "Thriller",
      "Crime"
    ],
    "runtime": 154
  },
];

export const MOCK_SORT_OPTIONS = [
  {
    name: 'Release date',
    value: 'release_date'
  },
  {
    name: 'Title',
    value: 'title'
  },
];

export const MOCK_GENRES = [
  'Crime',
  'Documentary',
  'Horror',
  'Comedy'
];
