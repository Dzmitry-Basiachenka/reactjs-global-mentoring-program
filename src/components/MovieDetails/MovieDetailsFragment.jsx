import React from 'react';
import { createSearchParams, useLoaderData, useNavigate, useSearchParams } from 'react-router-dom';
import { CloseButton } from 'react-bootstrap';

import MovieDetails from './MovieDetails';

const MovieDetailsFragment = () => {
  const [searchParams] = useSearchParams();
  const navigate = useNavigate();

  const movie = useLoaderData();

  return (
    <div>
      <span
        data-testid='close-movie-details'
        className='float-end'
        onClick={() => navigate({ pathname: '/', search: `?${createSearchParams(searchParams)}` })}
      >
        <CloseButton />
      </span>
      <MovieDetails movie={movie} />
    </div>
  );
}

export default MovieDetailsFragment;
