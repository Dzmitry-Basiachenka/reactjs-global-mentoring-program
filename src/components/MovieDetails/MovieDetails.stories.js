import MovieDetails from './MovieDetails.jsx';
import { MOCK_MOVIES } from '../../mock/mockData.js';

export default {
  component: MovieDetails
};

export const Default = {
  args: {
    movie: MOCK_MOVIES[0],
  },
  parameters: {
  }
};
