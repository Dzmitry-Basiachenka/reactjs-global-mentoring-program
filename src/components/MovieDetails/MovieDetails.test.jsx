import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, render, screen } from '@testing-library/react'

import MovieDetails from './MovieDetails.jsx';
import { MOCK_MOVIES } from '../../mock/mockData.js';

describe('MovieDetails', () => {

  afterEach(cleanup);

  it('should match snapshot', () => {
    const movieDetails = renderer
      .create(
        <MovieDetails
          movie={MOCK_MOVIES[0]}
        />
      )
      .toJSON();

    expect(movieDetails).toMatchSnapshot();
  });

  it('render correctly', () => {
    const movie = MOCK_MOVIES[0];

    render(
      <MovieDetails
        movie={movie}
      />
    );

    const image = screen.getByTestId('movie-poster');
    expect(image.getAttribute('src')).toEqual(movie.poster_path);
    expect(image.getAttribute('alt')).toEqual(movie.title);

    expect(screen.getByTestId('movie-title')).toHaveTextContent(movie.title);
    expect(screen.getByTestId('movie-vote')).toHaveTextContent(movie.vote_average);
    expect(screen.getByTestId('movie-tagline')).toHaveTextContent(movie.tagline);
    expect(screen.getByTestId('movie-genres')).toHaveTextContent('Thriller, Crime');
    expect(screen.getByTestId('movie-year')).toHaveTextContent('1994');
    expect(screen.getByTestId('movie-runtime')).toHaveTextContent('2h 34min');
    expect(screen.getByTestId('movie-overview')).toHaveTextContent(movie.overview);
  });

});
