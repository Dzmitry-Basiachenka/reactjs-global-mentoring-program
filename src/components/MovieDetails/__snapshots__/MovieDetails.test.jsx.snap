// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails should match snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-3"
    >
      <img
        alt="Pulp Fiction"
        className="img-fluid"
        data-testid="movie-poster"
        src="https://cdn.europosters.eu/image/1300/posters/pulp-fiction-cover-i1288.jpg"
      />
    </div>
    <div
      className="col-md-9"
    >
      <div
        className="flexbox"
      >
        <span
          className="title"
          data-testid="movie-title"
        >
          Pulp Fiction
        </span>
        <span
          className="vote"
          data-testid="movie-vote"
        >
          8.3
        </span>
      </div>
      <p
        data-testid="movie-tagline"
      >
        Just because you are a character doesn't mean you have character.
      </p>
      <p
        data-testid="movie-genres"
      >
        Thriller, Crime
      </p>
      <p
        className="flexbox"
      >
        <span
          className="year"
          data-testid="movie-year"
        >
          1994
        </span>
        <span
          className="runtime"
          data-testid="movie-runtime"
        >
          2h 34min
        </span>
      </p>
      <p
        data-testid="movie-overview"
      >
        A burger-loving hit man, his philosophical partner, a drug-addled gangster's moll and a washed-up boxer converge in this sprawling, comedic crime caper. Their adventures unfurl in three stories that ingeniously trip back and forth in time.
      </p>
    </div>
  </div>
</div>
`;
